src/librustc_typeck/check/_match.rs

```diff
@@ -659,6 +659,12 @@ pub fn check_pat_enum<'a, 'tcx>(pcx: &pat_ctxt<'a, 'tcx>,
     let report_bad_struct_kind = |is_warning| {
         bad_struct_kind_err(tcx.sess, pat.span, path, is_warning);
         if is_warning {
+            // Boo! Too painful to attach this to the actual warning,
```

Comment:
I debated about doing this but decided I was better off keeping this PR minimal to ease backporting to beta... what do you think?